### ISSUE TITLE
Question about the gt pose of the DeepFluoro dataset

### ISSUE THREAD

**Issue by yewentao2000** [eigenvivek/DiffDRR#208]
Dear eigenvivek, I would like to express my sincere thanks for your excellent work. I'm currently using the DeepFluoro database to create DRR in preparation for the database in my work. Unfortunately I found that I encountered resistance in the process of using the groundtruth pose to make the DRR image.
Specifically, I calculated the transformation matrix from the projection center to the pelvis from the extrinsic matrix according to the database instructions: 
`proj_to_pelvis = extrinsic * inverse(cam-to-pelvis-vol) `
Howerver I found that a huge. But I found that this pose does not help me simulate projection using Diffdrr because the dxdydz translation is unusually large ('dx': -69.41536, 'dy': -883.4053, 'dz': -803.965 for example in 17-1882/001). 
I looked into the code you are using in Diffpose and found that you preprocessed the data in this part
`        specimen.translate.compose(specimen.flip_xz)
        .compose(specimen.extrinsic.inverse())
        .compose(pose)
        .compose(specimen.lps2volume.inverse())`
I wonder what the meaning behind these operations and what coordinate system is the final transformation matrix based on? 
I would really appreciate it if you could give me a brief explanation!

**Comment by eigenvivek** [comment]
Hi @yewentao2000 thank you for your kind words. The core issue is that DiffDRR and DeepFluoro use different coordinate systems. The string of preprocessing transforms that you found move a pose from DeepFluoro's coordinate system to DiffDRR's.

A general overview is in Appendix D of the DiffPose paper (https://arxiv.org/pdf/2312.06358.pdf):
<img width="686" alt="image" src="https://github.com/eigenvivek/DiffDRR/assets/29757116/b96b6ff5-a0ae-49bd-8bf5-5aec7fe0bd87">

Happy to go into more detail if it would be helpful to you.

**Comment by yewentao2000** [comment]
Thank you for your timely and helpful reply!
After posting the question yesterday, I transplanted the coordinate transformation part in your code into my script and successfully completed the restoration of the Xray in the provieded pose. Through the analysis of the `compose` function, I converted your code into the following matrix multiplication form:
\$$pose = T_{VL}^{-1}\cdot T_{VW}\cdot T_{CW}^{-1} \cdot T_{flip}\cdot T_{CD}\$$
where:
\$T_{VL}\$ : LPS coordinate system to Volume coordinate sysytem transform matrix
\$T_{VW}\$ : World coordinate system to Volume coordinate sysytem transform matrix
\$T_{CW}\$ : World coordinate system to Camera coordinate sysytem transform matrix (`extrinsic` in DeepFluoro database)
\$T_{flip}\$ : flip transform matrix
\$T_{CD}\$ : Detector(DRR) coordinate system to Camera coordinate sysytem transform matrix (`translate` in diffpose code)

So after matrix multiplication we will get the transform matrix \$T_{LO}\$ , which represents the coordinate transformation from the DRR detector to the LPS coordinate system. 
This derivation means that the final `pose` obtained expresses the relative positional relationship between the detector and CT voxels. That is to say, in the subsequent 3D visualization process, we refer to this detector coordinate system (the origin is on the 1536*1536 pixel plane center) to establish a spatial relationship. Is my derivation correct?
I have been studying the data preprocessing stage for a long time due to my poor knowledge, please allow me to express my sincere thanks to you again!

**Comment by eigenvivek** [comment]
Yes you're correct! Good job parsing it, figuring out camera matrix conventions is really hard / annoying to keep track of. I spent the first month of the project just trying to get the camera matrix right :)

**Comment by yewentao2000** [comment]
Very thank for your reply! My last question is after going through the coordinate transformation matrix, I try to restore it back to the form of `roll pitch yaw dx dy dz`. I found that both `dx` and `dy` are relatively large (three digits), I guess this is because they are using pixel coordinates? In real space it requires `*spacing` to get units in millimeters?

**Comment by eigenvivek** [comment]
no they're in world coordinates - it's because the origin of the CT is waaaaay off in space.

in the development version of diffdrr, i've changed the geometry such that the isocenter of the ct is at the origin of world coordinates. this makes dx and dy much smaller.

still working on incorporating this change into diffpose, so you can expect a pretty big update to the geometry in 1-2 weeks.

**Comment by yewentao2000** [comment]
getcha! very thank for your patient response!
